Foundation/Operation.swift

```diff
@@ -25,6 +25,12 @@ open class Operation : NSObject {
     internal var _depGroup = DispatchGroup()
     internal var _groups = [DispatchGroup]()
 #endif
+    // prioritizedPrevious && prioritizedNext used to traverse in the order in which the operations need to be dequeued
+    fileprivate weak var prioritizedPrevious: Operation?
+    fileprivate var prioritizedNext: Operation?
+    // orderdedPrevious && orderdedNext used to traverse in the order in which the operations added to the queue
+    fileprivate weak var orderdedPrevious: Operation?
+    fileprivate var orderdedNext: Operation?
```

Comment:
Out of these four variables only one needs to be strong (or none if you balanced via the finish which you are going to have to anyhow)

The others really should be Unmanaged. But I guess that can be for a future enhancement.

Reply:
Thanks @phausler for the approval. 

Sure, I will think about more and find a way reduce the strong reference down to just one. I will send one more PR with that improvement, hope this implementation ok to merge.

Reply:
Resolving this comment since it with be addressed in future PR